vdb/extensions/arm.py

```diff
@@ -23,27 +41,41 @@ def thumb(db, line):
     if argc > 1:
         size = t.parseExpression(argv[1])
 
-    bytes = t.readMemory(addr, size)
+    bytez = t.readMemory(addr, size)
     offset = 0
 
     db.vprint("Dissassembly:")
     while offset < size:
         va = addr + offset
-        op = d.disasm(bytes, offset, va)
-        obytes = bytes[offset:offset+len(op)]
+        op = disasmobj.disasm(bytez, offset, va)
+        obytez = bytez[offset:offset+len(op)]
 
         db.canvas.addVaText('0x%.8x' % va, va=va)
-        db.canvas.addText(": %s " % e_common.hexify(obytes).ljust(17))
+        db.canvas.addText(": %s " % e_common.hexify(obytez).ljust(17))
         op.render(db.canvas)
         db.canvas.addText("\n")
 
         offset += len(op)
 
+def togglethumb(db, line):
+    '''
+    Toggle Thumb Mode
+    '''
+    t = db.getTrace()
+    cur_t = t.getRegister(e_arm_regs.REG_T)
+    new_t = not cur_t
+    arch = (envi.ARCH_ARMV7, envi.ARCH_THUMB)[new_t]
+    t.setRegister(e_arm_regs.REG_T, new_t)
+
+    db.canvas.addText("Toggled Thumb Mode: %r\n" % new_t)
+
 def vdbExtension(vdb, trace):
     vdb.addCmdAlias('db', 'mem -F bytes')
     vdb.addCmdAlias('dw', 'mem -F u_int_16')
     vdb.addCmdAlias('dd', 'mem -F u_int_32')
     vdb.addCmdAlias('dq', 'mem -F u_int_64')
     vdb.addCmdAlias('dr', 'mem -F "Deref View"')
     vdb.addCmdAlias('ds', 'mem -F "Symbols View"')
-    vdb.registerCmdExtension(thumb)
+    vdb.registerCmdExtension(armdis)
+    vdb.registerCmdExtension(thumbdis)
+    vdb.registerCmdExtension(togglethumb)
```

Comment:
Hmm. What do you think about a singular command that handles the disassembly and thumb state? We can keep the other two around, but having like a "dis <foo address>" for arm that just takes in the state of the thumb mode register and chooses the right disasm object without the user having to specify which one they want (that way most users only have to remember the one).

It'd be a minor convenience object, but I think it'd be nice. Or does it already exist and I just missed it?

Reply:
so, `do_dis` is a core VdbCli function.  
it currently uses the current setting for Tmode... but since we may want to disassemble code in some area that is the opposite mode from our current.

lol.  see, i keep hitting into other bugs solved by the other PR.

```
vdb > dis
Dissassembly:
pc_sys  0x00021104: 00b5              push {lr}
2021-11-02 09:55:38,700:ERROR:envi.memcanvas: Traceback (most recent call last):
  File "vivisect/envi/memcanvas/__init__.py", line 333, in _canvasCleared
    rsize = rend.render(self, va)
  File "vivisect/vdb/renderers.py", line 99, in render
    op.render(mcanv)
  File "vivisect/envi/archs/arm/disasm.py", line 4036, in render
    oper.render(mcanv, self, i)
  File "vivisect/envi/archs/arm/disasm.py", line 4795, in render
    value = self.getOperValue(op, mcanv.mem)
  File "vivisect/envi/archs/arm/disasm.py", line 4749, in getOperValue
    ret = emu.readMemValue(addr, self.tsize)
  File "vivisect/envi/memory.py", line 178, in readMemValue
    return e_bits.parsebytes(bytes, 0, size, False, self.getEndian())
AttributeError: 'TraceEmulator' object has no attribute 'getEndian'
[__init__.py:_canvasCleared:338:Thread-1]
        0x00021106: 044b              ldr r3,[0x00021118]
Render Exception At 0x21106: 'TraceEmulator' object has no attribute 'getEndian'
```